src/v/cluster/controller_backend.cc

```diff
@@ -663,7 +663,9 @@ void controller_backend::process_delta(const topic_table::ntp_delta& d) {
     // update partition_leaders_table if needed
 
     if (d.type == topic_table_ntp_delta_type::removed) {
-        _partition_leaders_table.local().remove_leader(d.ntp, d.revision);
+        ssx::spawn_with_gate(_gate, [this, ntp = d.ntp, rev = d.revision] {
+            return _partition_leaders_table.local().remove_leader(ntp, rev);
+        });
```

Comment:
will this lead to inconsistent behavior during replay because operations wont' apply in order?

Reply:
Not really, the updated will be queued in the semaphore waiters queue and executed in order